### PR TITLE
Feature: Filter navigation from aggregated report

### LINKED ISSUE #737: CLI: Filtered Navigations & Uniform filtering across URLs in an aggregated analysis
## Feature Request

**Description:**

Allow users to navigate from the Insights page to the detailed views for each analyzed URL: when navigating from  a given section in the Insights page, the user lands on the cookie table for the top frame of the first URL, with the filter options capturing the selected insights section. The same filter selection is uniformly applied to all frames of all analyzed URLs. The user can pick one section at a time from the Insights page, and the filters are "hydrated" appropriately across navigations.

### PR DESCRIPTION
**Pull request by mayan-000**
## Description
This PR enables navigating from a sitemap report insights page to the first site top frame when any filter is selected.

## Relevant Technical Choices
- As the category filter contains only 4 options, we will not calculate them from data, but keep them static in code.

## Testing Instructions
- Build CLI and analyze a sitemap.
- Open the HTML file and click on the Functional category on the sitemap report's landing page.
- It should open the first site's top frame with a functional filter selected.
- Now navigate to other sites' different frames, and the filter should persist.
- Try other filter selections, the navigation, and persistence should persist across the sites and frames.
- Now analyze a single site and check whether everything works correctly with the same scenarios.

## Screenshot/Screencast

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/58820001/1a19f50e-772f-4cd0-849e-f00aa01eee89

---

## Checklist

- [x] I have thoroughly tested this code to the best of my abilities.
- [x] I have reviewed the code myself before requesting a review.
- [x] This code is covered by unit tests to verify that it works as intended.
- [x] The QA of this PR is done by a member of the QA team (to be checked by QA).

Fixes #737

### CODE REVIEW

File: .github/workflows/lint-files.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@ jobs:
     timeout-minutes: 20
     steps:
       - name: Harden Runner # Security agent: blocks outbound traffic & detects code overwrite to prevent breaches
-        uses: step-security/harden-runner@128a63446a954579617e875aaab7d2978154e969
+        uses: step-security/harden-runner@17d0e2bd7d51742c71671bd19fa12bdc9d40a3d6
```
Comment (amovar18) [inline .github/workflows/lint-files.yml:45]: This is already in develop. Lets merge that into this and remove this unrelated diff.

**Review by mohdsayed — APPROVED** [review]
LGTM
